src/__tests__/starWarsData.js

```diff
@@ -76,29 +80,63 @@ const artoo = {
 };
 
 const droidData = {
-  2000: threepio,
-  2001: artoo,
+  '2000': threepio,
+  '2001': artoo,
 };
 
 /**
+ * These are data types corresponding to the schema design.
+ * They represent what the input and output datas should be
+ * in schema's resolving hierarchy.
+ * If the schema is re-designed,these types should be re-designed too.
+ */
+export type Character = {
+  id: string,
+  name?: string,
+  friends?: Array<string>,
+  appearsIn?: Array<number>,
+};
+
+export type Human = {
+  id: string,
+  name?: string,
+  friends?: Array<string>,
+  appearsIn?: Array<number>,
+  homePlanet?: string,
+};
+
+export type Droid = {
+  id: string,
+  name?: string,
+  friends?: Array<string>,
+  appearsIn?: Array<number>,
+  primaryFunction?: string
+};
+/**
  * Helper function to get a character by ID.
  */
-function getCharacter(id) {
+function getCharacter(id): Promise<*> {
```

Comment:
Typed all functions is some personal code style of me. Will remove this to fit the project's code style.